src/main/java/io/kestra/plugin/microsoft365/sharepoint/Create.java

```diff
@@ -0,0 +1,182 @@
+            title = "Create a folder in SharePoint",
+            full = true,
+            code = """
+                id: microsoft365_sharepoint_create_folder
+                namespace: company.team
+
+                tasks:
+                  - id: create_folder
+                    type: io.kestra.plugin.microsoft365.sharepoint.Create
+                    tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                    clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                    clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                    siteId: "contoso.sharepoint.com,2C712604-1370-44E7-A1F5-426573FDA80A,2D2244C3-251A-49EA-93A8-39E1C3A060FE"
+                    driveId: "b!BCTBCKSP50iysCOFPU"
+                    parentId: "01BYE5RZ6QN3ZWBTURF3F43DSUNZYRZD5Q"
+                    name: "NewFolder"
+                """
+        ),
+        @Example(
+            title = "Create a file with content in SharePoint",
+            full = true,
+            code = """
+                id: microsoft365_sharepoint_create_file
+                namespace: company.team
+
+                tasks:
+                  - id: create_file
+                    type: io.kestra.plugin.microsoft365.sharepoint.Create
+                    tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                    clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                    clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                    siteId: "contoso.sharepoint.com,2C712604-1370-44E7-A1F5-426573FDA80A,2D2244C3-251A-49EA-93A8-39E1C3A060FE"
+                    driveId: "b!BCTBCKSP50iysCOFPU"
+                    parentId: "01BYE5RZ6QN3ZWBTURF3F43DSUNZYRZD5Q"
+                    name: "document.txt"
+                    content: "Hello, SharePoint!"
+                """
+        )
+    }
+)
+@Schema(
+    title = "Create a file or folder in SharePoint.",
+    description = "Creates a new file with optional content or an empty folder in a SharePoint document library."
+)
+public class Create extends AbstractSharepointTask implements RunnableTask<Create.Output> {
+
+    @Schema(
+            title = "Parent folder ID",
+            description = "The ID of the parent folder where the item will be created. Use 'root' for the root of the document library."
+    )
+    @NotNull
+    private Property<String> parentId;
+
+    @Schema(
+        title = "The name of the item.",
+        description = "The name of the file or folder to create."
+    )
+    @NotNull
+    private Property<String> name;
+
+    @Schema(
+            title = "Item type",
+            description = "Whether to create a FILE or FOLDER"
+    )
+    @NotNull
+    @Builder.Default
+    private Property<ItemType> itemType = Property.ofValue(ItemType.FILE);
+
+    @Schema(
+        title = "The content of the file.",
+        description = "The content to be written to the new file. If not provided, an empty folder will be created."
+    )
+    private Property<String> content;
```

Comment:
Hmm I'm not sure it's very easy to figure it out for the user experience.
I'd like to have your opinion on it @anna-geller 🙏

Reply:
Well actually since we have an `itemType` property LGTM after all!